### DISCUSSION TITLE
Fabric API not working?

### DISCUSSION THREAD

**Discussion by 12122234** [FabricMC/fabric-api#2508] [Mod Dev Support] [answered]
i was working on a mod but for some reason when i launch the game a message pops up : 
mod "mod name" needs fabric API which is missing.
and i have fabric API in my mods folder and its the right version of it...
what can i do please?

**Comment by apple502j** [comment]
Just to check again:
The latest version for 1.19.2 is 0.60.0.
The latest version for 1.19.1 is 0.58.5.
The latest version for 1.19.0 is 0.58.0.

Please make sure to use the correct version for the game version (for ALL MODS). For example, installing a mod for 1.19.1 on 1.19 game generally does not work, and vice versa.

**Reply by 12122234** [reply]
i rechecked everything and its allright
fabric API version is 0.58.5
minecraft version 1.19.1
the mod is compatible with 1.19.1

**Reply by apple502j** [reply]
Can you send a list of mods?

**Reply by 12122234** [reply]
i just deleted and im reworking on my mod
i will give you the list of the other mods i downloaded from the browser : 
-fabric-1.19.1
-fabric-api-0.58.5+1.19.1
-fabric-loader-master
-fabric-loom-dev-1.0
-modmenu-4.0.5
-optifabric-1.13.10
-OptiFine_1.19.1_HD_U_H9

**Reply by 12122234** [reply]
i just recreated the mod and this is the error : 
`Incompatible mod set!
net.fabricmc.loader.impl.FormattedException: Mod resolution encountered an incompatible mod set!
A potential solution has been determined:
	 - Install fabric-api, any version.
Unmet dependency listing:
	 - Mod 'A Mod' (mod) ${version} requires any version of fabric-api, which is missing!
	at net.fabricmc.loader.impl.FabricLoaderImpl.load(FabricLoaderImpl.java:190)
	at net.fabricmc.loader.impl.launch.knot.Knot.init(Knot.java:148)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:68)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
	at net.fabricmc.devlaunchinjector.Main.main(Main.java:86)
`

**Reply by apple502j** [reply]
Try using the one from https://github.com/FabricMC/fabric-example-mod/tree/b242df2716bf03fdb13b64980c5c6d96489437e9

**Reply by 12122234** [reply]
it just got worse...
by using the link i wasn't able to create the mod because of way too many errors
have you got any other solution?

**Comment by warjort** [comment, marked as answer]
You need to show the actual errors. Preferably the full `run/logs/latest.log`
If you just say "it does work" with no further information, there is not much we can do to help you.

You can also put your project on github so we can look at it (and try it for ourselves if necessary).
Otherwise we have to guess which of the many possible problems it could be.

As a simple test, you can clone, build and run the example mod. Don't modify it.
https://github.com/FabricMC/fabric-example-mod
If you can't do that then there is some problem with your build environment, e.g. something broken in your gradle cache.

**Reply by 12122234** [reply]
Is a tutorial for 1.16.5 outdated?
I mean I'm following a 1.16.5 tutorial and putting the right values for 1.19.1 in gradle.properties file.
Please correct me if I'm wrong by doing that

**Reply by 12122234** [reply]
Now I did not modify the folder but only the gradle.properties file to match the game version but an error showed up: 
 `Incompatible mod set!
net.fabricmc.loader.impl.FormattedException: Mod resolution encountered an incompatible mod set!
A potential solution has been determined:
	 - Install fabric-api, any version.
Unmet dependency listing:
	 - Mod 'Example Mod' (modid) ${version} requires any version of fabric-api, which is missing!
	at net.fabricmc.loader.impl.FabricLoaderImpl.load(FabricLoaderImpl.java:190)
	at net.fabricmc.loader.impl.launch.knot.Knot.init(Knot.java:148)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:68)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
	at net.fabricmc.devlaunchinjector.Main.main(Main.java:86)
`

**Reply by warjort** [reply]
I would recommend you use more up-to-date examples. 

If you are just beginning modding, you should start from code you know works. 
Otherwise you won't know which are your errors and which are due to changes in recent versions of minecraft.

**Reply by 12122234** [reply]
I was searching for a 1.19.1 tutorial but there was none and that was a big problem for me I was forced to watch a 1.16.5 tutorial

**Reply by warjort** [reply]
````Now I did not modify the folder but only the gradle.properties file to match the game version````

Don't modify anything. 
The idea of trying to build the example mod is to use this known working build to test if your build environment is broken is some way.

**Reply by 12122234** [reply]
This time I downloaded the example mod and directly put it in the mods folder, and it did not show up in the mod menu...

**Reply by 12122234** [reply]
[fabric-example-mod-1.19.zip](https://github.com/FabricMC/fabric/files/9501474/fabric-example-mod-1.19.zip)

**Reply by warjort** [reply]
I don't understand what you are doing?

You don't put mods in the mods folder in development.
e.g. this is Modern Industrialisation using AE2 for testing:

Define the maven repository:
https://github.com/AztechMC/Modern-Industrialization/blob/f3374ad8210b68d2fcd83bbc77ca130dd3345c2e/build.gradle#L34
Include it as a runtime dependency
https://github.com/AztechMC/Modern-Industrialization/blob/f3374ad8210b68d2fcd83bbc77ca130dd3345c2e/build.gradle#L216

**Reply by warjort** [reply]
What I wanted you to do is test if you can build and run the example mod.
The project, not as a mod in another build.

But it sounds like you can build and run something else if you can see the mod menu?

**Reply by warjort** [reply]
That gradle project in that zip file you attached builds and runs fine for me.
Does it work for you?

**Reply by 12122234** [reply]
Yes, I'm able to add an item now!
Thanks for help
